src/Administration/Controller/ScriptController.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Administration\Controller;
+
+use Shopware\Administration\Framework\Script\AdminHook;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Script\Api\ScriptResponseEncoder;
+use Shopware\Core\Framework\Script\Execution\ScriptExecutor;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @internal
+ * @internal
+ */
+#[Route(defaults: ['_routeScope' => ['administration']])]
+#[Package('administration')]
+class ScriptController
+{
+    public const ROUTE = 'admin.script_endpoint';
+
+    public function __construct(
+        private readonly ScriptResponseEncoder $scriptResponseEncoder,
+        private readonly ScriptExecutor $scriptExecutor
+    ) {
+    }
+
+    #[Route(path: '/admin/app/{hook}', name: self::ROUTE, requirements: ['hook' => '.+'], defaults: ['auth_cookie' => true], methods: ['GET', 'POST'])]
+    public function execute(string $hook, Request $request, Context $context): Response
+    {
+        $hook = new AdminHook(
+            $hook,
+            $request->request->all(),
+            $request->query->all(),
+            $context
+        );
+
+        $content = $this->scriptExecutor->execute($hook);
+
+        $response = $this->scriptResponseEncoder->encodeByHook($hook, $request->get('includes', []));
+
+        // Allow pages in iframes
+        $response->headers->set('X-Frame-Options', 'sameorigin');
```

Comment:
IMHO this should be set by the script itself via the security service

Reply:
Good idea. I added it